### DISCUSSION TITLE
Tkinter python Travel Agency

### DISCUSSION THREAD

**Discussion by DavidMarkevich** [community/community#70133] [Programming Help]
### Select Topic Area

Question

### Body

I don't understand what's wrong with the code, when opened, it calls the tkinter window, but it doesn't show anything

import tkinter as tk
import tkinter.ttk as ttk
from tkinter import messagebox

# Главное окно приложения
root = tk.Tk()
root.title("Туристическое агентство")

# Создание вкладок
tab_control = ttk.Notebook(root)

# Вкладка "О нас"
about_tab = tk.Frame(tab_control)
tab_control.add(about_tab, text="О нас")

about_text = """
Мы - туристическое агентство, специализирующееся 
на продаже туров в различные страны.
У нас можно найти лучшие предложения по доступным ценам.
"""

# Функция для отображения информации о выбранной поездке
def show_trip_info(trip):
    trip_info = trip_data[trip]
    messagebox.showinfo(trip, f"Описание: {trip_info['Описание']}\nЦена: {trip_info['Цена']}")

about_label = tk.Label(about_tab, text=about_text, pady=10)
about_label.pack()

# Вкладка "Тур поездки"
trip_tab = tk.Frame(tab_control)
tab_control.add(trip_tab, text="Тур поездки")

# Глобальные переменные для сохранения данных о поездках
trip_data = {
    "Норвегия": {"Описание": "Норвегия - великолепная страна с красивой природой.", "Цена": "$1000"},
    "Ирландия": {"Описание": "Ирландия - зеленый рай.", "Цена": "$800"},
    "Швеция": {"Описание": "Швеция - современная страна со скандинавским колоритом.", "Цена": "$1200"}
}

# Функция для обработки оформления поездки
def book_trip(trip, departure_date, return_date, name):
    message = f"Поздравляем, {name}! Вы успешно оформили поездку в {trip}.\n" \
              f"Дата вылета: {departure_date}\nДата возвращения: {return_date}"
    messagebox.showinfo("Поездка оформлена", message)

# Создание кнопок для каждой поездки
for trip in trip_data:
    trip_button = tk.Button(trip_tab, text=trip,
                            command=lambda t=trip: show_trip_info(t))
    trip_button.pack(pady=5)

# Оформление поездки
booking_frame = tk.Frame(trip_tab)

departure_label = tk.Label(booking_frame, text="Дата вылета:")
departure_label.grid(row=0, column=0)

departure_entry = tk.Entry(booking_frame)
departure_entry.grid(row=0, column=1)

return_label = tk.Label(booking_frame, text="Дата возвращения:")
return_label.grid(row=1, column=0)

return_entry = tk.Entry(booking_frame)
return_entry.grid(row=1, column=1)

name_label = tk.Label(booking_frame, text="Фамилия и имя:")
name_label.grid(row=2, column=0)

name_entry = tk.Entry(booking_frame)
name_entry.grid(row=2, column=1)

book_button = tk.Button(booking_frame, text="Оформить поездку",
                        command=lambda: book_trip(tab_control.tab(tab_control.select(), "text"),
                                                  departure_entry.get(),
                                                  return_entry.get(),
                                                  name_entry.get()))
book_button.grid(row=3, columnspan=2, pady=10)

booking_frame.pack(pady=10)

# Вкладка "Отзывы"
reviews_tab = tk.Frame(tab_control)
tab_control.add(reviews_tab, text="Отзывы")

review_label = tk.Label(reviews_tab, text="Оставьте свой отзыв:")
review_label.pack(pady=10)

review_entry = tk.Entry(reviews_tab)
review_entry.pack(pady=5)

submit_button = tk.Button(reviews_tab, text="Отправить отзыв",
                          command=lambda: messagebox.showinfo("Отзыв отправлен",
                                                              "Спасибо за ваш отзыв!"))
submit_button.pack(pady=5)

# Отображение приложения

root.mainloop()

**Comment by Xrenya** [comment]
@DavidMarkevich I think you are missing the code line; `tab_control.pack()`.
